packages/router/src/route-callback.ts

```diff
@@ -0,0 +1,49 @@
+import { DI, IEventAggregator, Registration } from '@aurelia/kernel';
+import { batch } from '@aurelia/runtime';
+import { IWindow, AppTask } from '@aurelia/runtime-html';
+import { Navigation, RouterNavigationEndEvent } from '.';
+
+export type IRoute<TParams = Record<string, string | undefined>, TQueryParams = Record<string, string | undefined>> = {
+    title: string;
+    path: string;
+    url: string;
+    params: TParams;
+    queryParams: TQueryParams;
+    snapshot: Partial<Navigation>;
+  };
+
+export const IRoute = DI.createInterface<IRoute>();
```

Comment:
we need to decorate this with `/*@__PURE__*/` comment for tree shaking
```suggestion
export const IRoute = /*@__PURE__*/ DI.createInterface<IRoute>();
```